amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -220,77 +269,202 @@ describe('SearchPage', () => {
-      getSelectedTabByResourceTypeSpy = jest.spyOn(wrapper.instance(), 'getSelectedTabByResourceType').mockImplementation(() => {
-        return mockResourceType;
-      });
+    });
+
+    it('parses url params correctly', () => {
+      urlString = '/search?searchTerm=tableSearch&selectedTab=table&pageIndex=3';
+      urlParams = wrapper.instance().getUrlParams(urlString);
 
-      mockSelectedTab = ResourceType.table;
+      expect(urlParams.term).toEqual('tableSearch');
+      expect(urlParams.tab).toEqual('table');
+      expect(urlParams.index).toEqual(3);
+    });
+
+    it('parses url params correctly', () => {
+      urlString = '/search?searchTerm=userSearch&selectedTab=user&pageIndex=0';
+      urlParams = wrapper.instance().getUrlParams(urlString);
 
-      wrapper.instance().getSanitizedUrlParams('current', 0, mockSelectedTab)
+      expect(urlParams.term).toEqual('userSearch');
+      expect(urlParams.tab).toEqual('user');
+      expect(urlParams.index).toEqual(0);
     });
 
-    it('calls getSelectedTabByResourceType with correct value', () => {
-      expect(getSelectedTabByResourceTypeSpy).toHaveBeenCalledWith(mockSelectedTab);
+    it('trims empty spaces from searchTerm', () => {
+      urlString = '/search?searchTerm= term%20&selectedTab=user&pageIndex=0';
+      urlParams = wrapper.instance().getUrlParams(urlString);
+
+      expect(urlParams.term).toEqual('term');
+      expect(urlParams.tab).toEqual('user');
+      expect(urlParams.index).toEqual(0);
     });
 
-    it('output of getSanitizedUrlParams is expected', () => {
-      const expected = {'term': 'current', 'index': 0, 'currentTab': ResourceType.table};
-      expect(wrapper.instance().getSanitizedUrlParams('current', 0, ResourceType.table)).toEqual(expected);
+    it('defaults NaN pageIndex as 0', () => {
+      urlString = '/search?searchTerm=current&selectedTab=table&pageIndex=NotANumber';
+      urlParams = wrapper.instance().getUrlParams(urlString);
+
+      expect(urlParams.term).toEqual('current');
+      expect(urlParams.tab).toEqual('table');
+      expect(urlParams.index).toEqual(0);
     });
 
-    it('output of getSanitizedUrlParams is expected for undefined vars', () => {
-      const expected = {'term': '', 'index': 0, 'currentTab': ResourceType.table};
-      expect(wrapper.instance().getSanitizedUrlParams(undefined, undefined, ResourceType.table)).toEqual(expected);
+    it('defaults invalid tabs to the current tab', () => {
+      urlString = '/search?searchTerm=current&selectedTab=invalidTabType&pageIndex=0';
+      urlParams = wrapper.instance().getUrlParams(urlString);
+
+      expect(urlParams.term).toEqual('current');
+      expect(urlParams.tab).toEqual(props.selectedTab);
+      expect(urlParams.index).toEqual(0);
     });
+
   });
 
   describe('getSelectedTabByResourceType', () => {
     let props;
     let wrapper;
 
     beforeAll(() => {
-      const setupResult = setup();
+      const setupResult = setup(null, {
```

Comment:
Why does the setup need to be anything besides the default setup, when this method does not need to know anything about component props or state?

Reply:
Probably an oversight from merging.